### ISSUE TITLE
How to give permission based on database connections programatically

### ISSUE THREAD

**Issue by nagarajmmu** [apache/superset#24830]
#### How to reproduce the bug

1. Create DATABASE_1 connection and DATABASE_2 connection.
2. Create dataset for DATABASE_1 and create dataset for DATABASE_2.
3. If two user exist, one user should access dataset belongs to DATABASE_1
4. And second user should access only dataset belonging to DATABASE_2.
5. Is this possible programmatically assign database connection wise permission?
6. Note: DATABASE connections are creating dynamically through API, because in our case, more than 100 database connections will exist.

### Expected results

Assign particular database connection permission to particular user programmatically.

### Actual results

Not able to assign particular database connection permission to particular user programmatically. 

### Environment

(please complete the following information):

- superset version: v2.1.0
- python version: 3.8

### Additional context

We are using Superset with Duckdb, in duckdb, each duckdb file will connect as a each database connection, if we have 100 duckdb file, 100 database connections will create. each duckdb file will have different datasets, one user will have only access to dataset belongs to particular duckdb file. 
Right now we are able to assign one database permission to user manually in superset UI, but we need to make it dynamic by program.

Please le me know, if anyone have solution.

Thanks
Nagaraj M M

**Comment by sebastianliebscher** [comment]
Hi @nagarajmmu, this might help you to [create or update roles in Apache Superset programmatically](https://stackoverflow.com/a/76839571/22257868) via the REST API.

@sfirke as this is not a bug, I think this issue can be moved to discussions.

**Comment by sfirke** [comment]
Agreed @sebastianliebscher - I'll convert to a discussion. Thanks for providing the link & response! @nagarajmmu in addition to GitHub discussions there's also the Superset Slack chat if you have additional questions.
